packages/compass-query-bar/src/components/query-bar/query-bar.jsx

```diff
@@ -220,6 +221,7 @@ class QueryBar extends Component {
       this.props[option] : this.props[`${option}String`];
 
     const label = OPTION_DEFINITION[option].label || option;
+    const placeholder = option === 'sort' && sortOptionPlaceholder ? sortOptionPlaceholder : OPTION_DEFINITION[option].placeholder;
```

Comment:
We're totally getting close here, that's a much nicer implementation, but in a way have the same issue the previous implementation had, it only allows to override the placeholder for one specific filed which is not really intuitive and that's the main problem I see here.

If we are changing this part of the code, I think we might as well allow those placeholder overrides for all fields. Check out how the value and validation status are derived for the fields a few lines above:

https://github.com/mongodb-js/compass/blob/04db91c47662ba476cb9680d4ce22a33dfbbf9f5/packages/compass-query-bar/src/components/query-bar/query-bar.jsx#L209-L217

I think we might want to do something similar here for consistency
```suggestion
    const placeholder = this.props[`${option}Placeholder`] || OPTION_DEFINITION[option].placeholder;
```

Reply:
Got it!